### PR TITLE
added formula for spark 1.6

### PR DESCRIPTION
**Pull request by admackin**
There is a new major version of Spark out, so adding a formula for an up-to-date 1.x version (which will continue to have releases for a while – a 1.6.3 release is [planned](https://issues.apache.org/jira/browse/SPARK?selectedTab=com.atlassian.jira.jira-projects-plugin:roadmap-panel) )

- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-versions/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)?
- [x] Does your submission adhere to Versions "Acceptable Formulae" [guidelines](https://github.com/Homebrew/homebrew-versions/blob/07b17dfb9f1/README.md#acceptable-formulae)?

-----

### CODE REVIEW

File: apache-spark16.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    # Rename beeline to distinguish it from hive's beeline
+    mv "bin/beeline", "bin/spark-beeline"
+
+    rm_f Dir["bin/*.cmd"]
+    libexec.install Dir["*"]
+    bin.write_exec_script Dir["#{libexec}/bin/*"]
+  end
+
+  test do
+    system "#{bin}/spark-shell <<<'sc.parallelize(1 to 1000).count()'"
```
Comment (MikeMcQuaid) [inline apache-spark16.rb]: Does this accept arguments from files or pipes?

Comment (MikeMcQuaid) [inline apache-spark16.rb]: Actually, never mind 👍

**Comment by MikeMcQuaid** [comment]
Can you remove apache-spark141 in this PR? Otherwise looks good 👍

**Comment by admackin** [comment]
Sorry, don't quite follow? AFAICT, my PR only include apache-spark16.rb (or am I missing something?). Or do you mean add apache-spark141 in 'conflicts_with' (presumably also apache-spark15 in that case)

**Comment by heuermh** [comment]
+1

@MikeMcQuaid I created a separate pull request removing apache-spark-141 at #1356

**Comment by MikeMcQuaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

For future reference the preferred commit message format for new formulae is `apache-spark16 1.6.2 (new formula)`. Please complete the issue template in future PRs where this format is detailed in the linked [Contributing guidelines](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md). Thanks!

**Comment by heuermh** [comment]
Thank you, @admackin!

**Pull request merged**